lib/compiler-singlepass/src/machine_arm64.rs

```diff
@@ -3072,49 +3072,38 @@ impl Machine for MachineARM64 {
         Ok(())
     }
     fn i32_popcnt(&mut self, loc: Location, ret: Location) -> Result<(), CompileError> {
-        // no opcode for that.
-        // 2 solutions: using NEON CNT, that count bits per Byte, or using clz with some shift and loop
         let mut temps = vec![];
-        let src = self.location_to_reg(Size::S32, loc, &mut temps, ImmType::None, true, None)?;
-        let dest = self.location_to_reg(Size::S32, ret, &mut temps, ImmType::None, false, None)?;
-        let src = if src == loc {
-            let tmp = self.acquire_temp_gpr().ok_or_else(|| {
-                CompileError::Codegen("singlepass cannot acquire temp gpr".to_owned())
-            })?;
-            temps.push(tmp);
-            self.assembler
-                .emit_mov(Size::S32, src, Location::GPR(tmp))?;
-            Location::GPR(tmp)
-        } else {
-            src
-        };
-        let tmp = {
-            let tmp = self.acquire_temp_gpr().ok_or_else(|| {
-                CompileError::Codegen("singlepass cannot acquire temp gpr".to_owned())
-            })?;
-            temps.push(tmp);
-            Location::GPR(tmp)
-        };
-        let label_loop = self.assembler.get_label();
-        let label_exit = self.assembler.get_label();
-        self.assembler
-            .emit_mov(Size::S32, Location::GPR(GPR::XzrSp), dest)?; // 0 => dest
```

Comment:
We need to use this codepath if the chipset doesn't support NEON (fallback).
So please wrap the old / new code in a if/else

Reply:
see 4a43746